### ISSUE TITLE
The "component" mini-framework needs a review and refactoring pass

### ISSUE THREAD

**Issue by caprica** [caprica/vlcj#698]
See #54, #337, #161, #132.

**Comment by caprica** [comment]
Previously some of the lesser-used components were not on a par with others - EmbeddedMediaPlayer is the most likely one to be widely used - this refactoring pass has brought all components (such as MediaListPlayerComponent and so on) to the same standard.

All of the template methods invoked by constructors to configure the component have been removed - previously this has led to subtle bugs and confusing behaviour in client application code.

Component creation can now be done using a MediaPlayerComponentBuilder - this new component implements a type-safe builder pattern. Use of this component is encouraged, but it is still possible to create components directly - rather than using the template methods for configuration, all configurable aspects are passed via constructors.

All components are now implemented with a consistent pattern. The many empty event listener methods (designed to be easily overridden by client applications) were moved out of the components themselves and into private base classes (applications never see this, but it keeps the component code nice and simple and is easier to grok and maintain).

Components can either create their own MediaPlayerFactory or client applications can supply their own - if the component creates one itself, then that component will automatically release the factory when the component is released; if a factory is passed in, it is the responsibility of the client application to release the factory at the appropriate time.

In short, the API has changed, but the end result is much nicer now, with simpler and more clear code.

**Comment by caprica** [comment]
First pass completed at ddfdbc158e68bd74e64440cb9865d9e386d5e19b.

Javadoc changes and tests still need to be revisited. This is covered by other tickets.
